Source/ExcelDna.Loader/AssemblyManager.cs

```diff
@@ -36,77 +38,92 @@ internal static void Initialize(IntPtr hModule, string pathXll)
             }
+         
+            Logger.Initialization.Info("Attempting to load {0} from resources.", name);
 
-            // We expect failures for Resource assemblies
-            // From: http://blogs.msdn.com/b/suzcook/archive/2003/05/29/57120.aspx
-            // "Note: Unless you are explicitly debugging the failure of a resource to load, 
-            //        you will likely want to ignore failures to find assemblies with the ".resources" extension 
-            //        with the culture set to something other than "neutral". Those are expected failures when the 
-            //        ResourceManager is probing for satellite assemblies."
-            bool isResourceAssembly = name.EndsWith(".RESOURCES", StringComparison.InvariantCultureIgnoreCase) /*&& assName.CultureInfo.Name != "neutral"*/;
-
-            // Now check in resources ...
-            if (isResourceAssembly)
-                Logger.Initialization.Verbose("Attempting to load {0} from resources.", name);
-            else
-                Logger.Initialization.Info("Attempting to load {0} from resources.", name);
-
-			assemblyBytes = GetResourceBytes(name, 0);
-			if (assemblyBytes == null)
-			{
-                if (isResourceAssembly)
-                    Logger.Initialization.Verbose("Assembly {0} could not be loaded from resources (ResourceManager probing for satellite assemblies).", name);
+            ushort lcid = 0;
+            // if we don't have a culture info or of it is the Neutral culture take 0, otherwise take LCID
+            if (assName.CultureInfo != null &&  !string.IsNullOrEmpty(assName.CultureInfo.Name))
+                lcid = (ushort)assName.CultureInfo.TextInfo.LCID;
+            assemblyBytes = ResourceHelper.LoadResourceBytes(hModule, "ASSEMBLY", name, lcid);
+          
+            if (assemblyBytes == null)
+            {                
+                // if the missing Assembly only a Resource use a lower LogLevel
+                if (assName.Name.EndsWith(".RESOURCES", StringComparison.InvariantCultureIgnoreCase))
+                    Logger.Initialization.Verbose("Assembly {0} could not be loaded from resources.", name);
                 else
                     Logger.Initialization.Warn("Assembly {0} could not be loaded from resources.", name);
-				return null;
-			}
+                return null;
+            }
 
             Logger.Initialization.Info("Trying Assembly.Load for {0} (from {1} bytes).", name, assemblyBytes.Length);
-			//File.WriteAllBytes(@"c:\Temp\" + name + ".dll", assemblyBytes);
-			try
-			{
-				loadedAssembly = Assembly.Load(assemblyBytes);
-                Logger.Initialization.Info("Assembly Loaded from bytes. FullName: {0}", loadedAssembly.FullName);
-				loadedAssemblies.Add(name, loadedAssembly);
-				return loadedAssembly;
-			}
-			catch (Exception e)
-			{
+            //File.WriteAllBytes(@"c:\Temp\" + name + ".dll", assemblyBytes);
+
+            // compute the hash of the bytes
```

Comment:
@govert this codeline prevent 100% against loading more than one time from the packed resources, because it checks the real loaded bytes